### ISSUE TITLE
Incorrect @return type at getSourceModel in Eav\Attribute

### ISSUE THREAD

**Issue by pedrodim** [magento/magento2#25278]
### Preconditions (*)

1. Magento 2.3.x

### Steps to reproduce (*)

1. Check DocBlock of `\Magento\Catalog\Model\ResourceModel\Eav\Attribute::getSourceModel()` in file `vendor/magento/module-catalog/Model/ResourceModel/Eav/Attribute.php`.

### Expected result (*)

1. `getSourceModel` DocBlock should say it returns `string|null`.

```
    /**
     * Retrieve source model
     *
     * @return string|null
     */
    public function getSourceModel()
    {
```

### Actual result (*)

1. `getSourceModel` DocBlock return type is `\Magento\Eav\Model\Entity\Attribute\Source\AbstractSource`.

```
    /**
     * Retrieve source model
     *
     * @return \Magento\Eav\Model\Entity\Attribute\Source\AbstractSource
     */
    public function getSourceModel()
    {
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-22158` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by VladimirZaets** [comment]
Hi @pedrodim. Thank you for your report. 
The issue has been fixed in magento/magento2#25333 by @mkalakailo in 2.4-develop branch
Related commit(s):
 - [c1c06cc71657fe264c24f7cbf9c54a1adaa21a2c](https://github.com/magento/magento2/commit/c1c06cc71657fe264c24f7cbf9c54a1adaa21a2c)
 - [eb43ee7a71064ccb1e9fbdaeeaa89dd45187aef8](https://github.com/magento/magento2/commit/eb43ee7a71064ccb1e9fbdaeeaa89dd45187aef8)
 - [8f30b1e75f999695cde7fa0d53df5d6e7da4385f](https://github.com/magento/magento2/commit/8f30b1e75f999695cde7fa0d53df5d6e7da4385f)

The fix will be available with the upcoming 2.4.0 release.
